src/main/java/io/fabric8/jenkins/openshiftsync/OpenShiftUtils.java

```diff
@@ -351,6 +353,32 @@ public static boolean isCancelled(BuildStatus status) {
     return status != null && status.getCancelled() != null && Boolean.TRUE.equals(status.getCancelled());
   }
 
+  /**
+   * Checks to see if there's a registry name already provided in the image name
+   *
+   * Code influenced by io.fabric8.kubernetes.pipeline.devops.ApplyStepExecution, which was 
+   * influenced from <a href="https://github.com/rhuss/docker-maven-plugin/blob/master/src/main/java/org/jolokia/docker/maven/util/ImageName.java">docker-maven-plugin</a>
+   * @param imageName
+   * @return true if the image name contains a registry
+   */
+  public static boolean hasRegistry(String imageName) {
+      if (imageName == null) {
+          return false;
+      }
+      Pattern tagPattern = Pattern.compile("^(.+?)(?::([^:/]+))?$");
+      Matcher matcher = tagPattern.matcher(imageName);
+      if (!matcher.matches()) {
+          // if here then imageName is not a proper image name ([registry/][repo][:port]");
+          return false;
+      }
+
+      String rest = matcher.group(1);
+      String[] parts = rest.split("\\s*/\\s*");
+      String part = parts[0];
+
+      return part.contains(".") || part.contains(":");
+  }
```

Comment:
doesn't look quite right.  talk to the platform management team.  they own imagestreams.